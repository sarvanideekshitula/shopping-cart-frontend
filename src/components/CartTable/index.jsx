import React from 'react';
import PropTypes from 'prop-types';
import * as styles from './index.module.css';


const CartTable = (props) => {
  const { cart, categories, testId } = props;
  const categoryItems = [];
  categories.forEach((category) => {
    const items = cart.items.filter((item) => item.product.category === category);
    if (items.length) {
      categoryItems.push({ type: 'category', value: category });
      items.forEach((item) => categoryItems.push({ type: 'item', value: item }));
    }
  });
  const categoryList = categoryItems.map((categoryItem) => {
    if (categoryItem.type === 'category') {
      return (
        <tr><td colSpan="4">{categoryItem.value}</td></tr>
      );
    }
    return (
      <tr>
        <td>{categoryItem.value.product.name}</td>
        <td>{categoryItem.value.product.price}</td>
        <td>{categoryItem.value.quantity}</td>
        <td>{categoryItem.value.totalPrice}</td>

      </tr>
    );
  });
  return (
    <div className={styles.CartTable} data-testId={testId}>
      <table className={styles.table}>
        <tr>
          <th>Item</th>
          <th>Price</th>
          <th>Quantity</th>
          <th>SubTotal</th>
        </tr>
        {categoryList}
      </table>
    </div>
  );
};

CartTable.propTypes = {
  cart: PropTypes.objectOf.isRequired,
  categories: PropTypes.objectOf.isRequired,
  testId: PropTypes.string.isRequired,
};
export default CartTable;
