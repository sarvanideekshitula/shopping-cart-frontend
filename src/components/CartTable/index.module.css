table {
    border-collapse: collapse;
    border-radius: 5px;
    width: 50rem;
  }
  
  table, td, th {
    border: 0.1rem solid black;
    padding: 20px 20px 20px 20px;

  }
  .table {
    font-family: "Trebuchet MS", Arial, Helvetica, sans-serif;
    border-collapse: collapse;
    padding:10px 40px; 
  }

