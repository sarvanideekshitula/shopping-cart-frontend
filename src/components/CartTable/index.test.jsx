import React from 'react';
import { render } from '@testing-library/react';
import CartTable from './index';

describe('the Genre container', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<CartTable cart={{}} categories={[]} testId="test" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
