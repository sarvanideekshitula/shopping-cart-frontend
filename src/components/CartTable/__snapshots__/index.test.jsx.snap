// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Genre container should render correctly 1`] = `
<DocumentFragment>
  <div
    data-testid="test"
  >
    <table>
      <tbody>
        <tr>
          <th>
            Item
          </th>
          <th>
            Price
          </th>
          <th>
            Quantity
          </th>
          <th>
            SubTotal
          </th>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
